src/llama_stack_client/lib/agents/agent.py

```diff
@@ -35,7 +34,32 @@ def create_session(self, session_name: str) -> int:
         self.sessions.append(self.session_id)
         return self.session_id
 
-    async def create_turn(
+    def _has_tool_call(self, chunk):
+        if chunk.event.payload.event_type != "turn_complete":
+            return False
+        message = chunk.event.payload.turn.output_message
+        if len(message.tool_calls) == 0:
+            return False
+        if message.stop_reason == "out_of_tokens":
```

Comment:
this could go earlier and then the return could be `return len(message.tool_calls) > 0`